doc: openthread: Describe nRF radio implementation (#23129)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

Thread
------

* Added the new architecture option to use the OpenThread stack directly to communicate with the IEEE 802.15.4 radio driver.

>   See the :ref:`openthread_stack_architecture` user guide for more information.

Reviewer: I would like information in the changelog (about the size differences i.e. optimization in kB) - e.g. for representative sample which could be CLI.

Author: I will describe the percentage difference

Thread
------

* Added the new architecture option to use the OpenThread stack directly to communicate with the IEEE 802.15.4 radio driver.
  See the :ref:`openthread_stack_architecture` user guide for more information.

>   The new architecture option reduces the memory footprint of the OpenThread stack by around 4% and the RAM usage by around 12% on the representative :ref:`ot_cli_sample` sample.

Reviewer 2: ```suggestion
  The new architecture option reduces the memory footprint of the OpenThread stack by around 4% and the RAM usage by around 12% in the :ref:`ot_cli_sample` sample.
```

Thread samples
--------------

> * Switched to the new :ref:`architecture option <openthread_stack_architecture>` to use the OpenThread stack directly to communicate with the IEEE 802.15.4 radio driver in the following samples:

Reviewer 3: ```suggestion
* Added the new :ref:`architecture option <openthread_stack_architecture>` to use the OpenThread stack directly to communicate with the IEEE 802.15.4 radio driver in the following samples:
```
